app/views/layouts/application.html.haml

```diff
@@ -20,13 +20,20 @@
         = link_to app_name, root_path, class: 'navbar-brand'
         .navbar-collapse.collapse.navbar-responsive-collapse
           %ul.nav.navbar-nav
-            %li= link_to 'Recipes', recipes_path
             - if user_signed_in?
+              %li= link_to 'Recipes', recipes_path
               %li= link_to 'Logout', destroy_user_session_path, method: :delete, data: { confirm: 'Are you sure?' }
             - else
               %li= link_to 'Login', new_user_session_path
               %li= link_to 'Sign Up', new_user_registration_path
 
+          %ul.nav.navbar-nav.navbar-text.navbar-right
+            - if user_signed_in?
```

Comment:
I try to put above but it didn't work. Wasn't able to put the current user email on the right. :(